server/repository/src/migrations/views/mod.rs

```diff
@@ -195,57 +195,74 @@ pub(crate) fn rebuild_views(connection: &StorageConnection) -> anyhow::Result<()
     ORDER BY datetime, type_precedence;
 
   CREATE VIEW stock_line_ledger_discrepancy AS 
-  WITH 
-  allocated_not_picked AS (
-      SELECT stock_line_id,
-          SUM(number_of_packs * pack_size) AS q
-      FROM invoice_line
-          JOIN invoice on invoice.id = invoice_line.invoice_id
-      WHERE invoice_line.type = 'STOCK_OUT'
-          AND invoice.status IN ('NEW', 'ALLOCATED')
-      GROUP BY 1
-  ),
-  max_ledger_datetime AS (
-      SELECT stock_line_id,
-          MAX(datetime) AS dt
-      FROM stock_movement
-      GROUP BY 1
-  ),
-  running_balance AS (
-      SELECT stock_line_ledger.stock_line_id,
-          running_balance AS q
-      FROM stock_line_ledger
-          JOIN max_ledger_datetime on stock_line_ledger.stock_line_id = max_ledger_datetime.stock_line_id
-          AND stock_line_ledger.datetime = max_ledger_datetime.dt
-  ),
-  current_balance AS (
-      SELECT stock_line.id AS stock_line_id,
-          store_id,
-          available_number_of_packs * pack_size AS a_q,
-          total_number_of_packs * pack_size AS t_q
-      from stock_line
-  )
-  SELECT DISTINCT stock_line_id
-  FROM stock_line_ledger
-  WHERE running_balance < 0
+  WITH
+	allocated_not_picked AS (
+		SELECT
+			stock_line_id,
+			SUM(number_of_packs * pack_size) AS q
+		FROM
+			invoice_line
+			JOIN invoice ON invoice.id = invoice_line.invoice_id
+		WHERE
+			invoice_line.type = 'STOCK_OUT'
+			AND invoice.status IN ('NEW', 'ALLOCATED')
+		GROUP BY
+			1
+	),
+	max_ledger_datetime AS (
+		SELECT
+			stock_line_id,
+			MAX(datetime) AS dt
+		FROM
+			stock_movement
+		GROUP BY
+			1
+	),
+	running_balance AS (
+		SELECT
+			stock_line_ledger.stock_line_id,
+			running_balance AS q
+		FROM
+			stock_line_ledger
+			JOIN max_ledger_datetime ON stock_line_ledger.stock_line_id = max_ledger_datetime.stock_line_id
+			AND stock_line_ledger.datetime = max_ledger_datetime.dt
+	),
+	current_balance AS (
+		SELECT
+			stock_line.id AS stock_line_id,
+			available_number_of_packs * pack_size AS a_q,
+			total_number_of_packs * pack_size AS t_q
+		FROM
+			stock_line
+	)
+  SELECT DISTINCT
+    stock_line_id
+  FROM
+    stock_line_ledger
+  WHERE
+    stock_line_ledger.running_balance < 0
   UNION
-  SELECT running_balance.stock_line_id
-  FROM running_balance
-      JOIN current_balance ON running_balance.stock_line_id = current_balance.stock_line_id
-      LEFT JOIN allocated_not_picked ON running_balance.stock_line_id = allocated_not_picked.stock_line_id
-  WHERE NOT(
-          running_balance.q = current_balance.t_q
-          AND (
-              (
-                  allocated_not_picked.q IS NULL
-                  AND current_balance.t_q = current_balance.a_q
-              )
-              OR (
-                  allocated_not_picked.q IS NOT NULL
-                  AND current_balance.a_q + allocated_not_picked.q = current_balance.t_q
-              )
-          )
-      );
+  SELECT
+    current_balance.stock_line_id
```

Comment:
The key here is selecting from current_balance and joining on running_balance instead of the other way around. And the `OR running_balance.q IS NULL;` at the end

Reply:
There has been a change in 2.11 to move view to their own files, this change has the potential of being lost in conflicts resolution

Reply:
I was hoping to just resolve the conflicts proper? I would say that the tests should fail but I didn't actually add a test for this - I should!

Reply:
now we have https://github.com/msupply-foundation/open-msupply/pull/9204/commits/0dabdf46ddfc702e95f22a72603136151bb0f944